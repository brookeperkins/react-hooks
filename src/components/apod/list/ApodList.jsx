import React from 'react';
import PropTypes from 'prop-types';

const ApodList = ({ title, url, explanation }) => (
  <figure>
    <h1>{title}</h1>
    <h3>{url}</h3>
    <h3>About: {explanation}</h3>
  </figure>
);

ApodList.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  explanation: PropTypes.string.isRequired
};

export default ApodList; 
