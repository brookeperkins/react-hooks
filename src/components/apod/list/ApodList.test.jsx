import React from 'react';
import { render } from '@testing-library/react';
import ApodList from './ApodList';

describe('ApodList component', () => {
  it('renders the ApodList component', () => {
    const { asFragment } = render(<ApodList 
      title="Messier 76"
      url="https://apod.nasa.gov/apod/image/0611/Messier76_seip_smallcc.jpg"
      explanation="Nebula at the right foot of Andromeda..."
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
