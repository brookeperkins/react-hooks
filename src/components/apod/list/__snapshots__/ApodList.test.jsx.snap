// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApodList component renders the ApodList component 1`] = `
<DocumentFragment>
  <figure>
    <h1>
      Messier 76
    </h1>
    <h3>
      https://apod.nasa.gov/apod/image/0611/Messier76_seip_smallcc.jpg
    </h3>
    <h3>
      About: Nebula at the right foot of Andromeda...
    </h3>
  </figure>
</DocumentFragment>
`;

exports[`Character component renders a character 1`] = `
<DocumentFragment>
  <figure>
    <h1>
      Messier 76
    </h1>
    <h3>
      https://apod.nasa.gov/apod/image/0611/Messier76_seip_smallcc.jpg
    </h3>
    <h3>
      About: Nebula at the right foot of Andromeda...
    </h3>
  </figure>
</DocumentFragment>
`;
